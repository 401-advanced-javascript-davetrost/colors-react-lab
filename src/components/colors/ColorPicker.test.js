import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('Color Picker component', () => {
  const colors = ['red', 'green', 'yellow'];

  it('renders buttons labelled red, green and yellow', () => {
    const wrapper = shallow(
      <ColorPicker
        selectColor={() => null}
        colors={colors} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('captures a click on the green color button', () => {
    let testColor;
    const wrapper = shallow(
      <ColorPicker
        selectColor={(color) => testColor = color}
        colors={colors} />
    );
    wrapper.findWhere(el => el.key() === 'green').simulate('click');
    expect(testColor).toBe('green');
  });
});
