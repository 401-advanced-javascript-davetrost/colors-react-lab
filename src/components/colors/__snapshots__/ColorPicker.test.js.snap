// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Picker component renders buttons labelled red, green and yellow 1`] = `
<Fragment>
  <button
    key="red"
    onClick={[Function]}
  >
    red
  </button>
  <button
    key="green"
    onClick={[Function]}
  >
    green
  </button>
  <button
    key="yellow"
    onClick={[Function]}
  >
    yellow
  </button>
</Fragment>
`;
